mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndex.java

```diff
@@ -213,4 +217,40 @@ private void logIfNoMessageId(SearchResult searchResult) {
             LOGGER.error("No messageUid for {} in mailbox {}", searchResult.getMessageUid(), searchResult.getMailboxId());
         }
     }
+
+    @Override
+    public Mono<Flags> retrieveIndexedFlags(Mailbox mailbox, MessageUid uid) {
+        RoutingKey routingKey = routingKeyFactory.from(mailbox.getMailboxId());
+
+        return elasticSearchIndexer.get(indexIdFor(mailbox, uid), routingKey)
+            .filter(GetResponse::isExists)
+            .map(GetResponse::getSourceAsMap)
+            .map(this::extractFlags)
+            .switchIfEmpty(Mono.error(new IndexNotFoundException(
+                String.format("Index for message %s in mailbox %s not found", uid.toString(), mailbox.getMailboxId().serialize()))));
+    }
+
+    private Flags extractFlags(Map<String, Object> source) {
+        Flags flags = new Flags();
+
+        if ((Boolean) source.get(JsonMessageConstants.IS_ANSWERED)) {
+            flags.add(Flags.Flag.ANSWERED);
+        }
+        if ((Boolean) source.get(JsonMessageConstants.IS_DELETED)) {
+            flags.add(Flags.Flag.DELETED);
```

Comment:
This mutability is not nice. Use `FlagsBuilder` instead.

I propose to modify it to also allow this syntax:

```
return FlagsBuilder.builder()
    .isAnswered(boolean)
    .isDeleted(boolean)
    .isDraft(boolean)
    ...
    .build();
```